lib/monitor.sh

```diff
@@ -0,0 +1,56 @@
+
+monitor_memory_usage() {
+  local output_file="$1"
+
+  # drop the first argument, and leave other arguments in place
+  shift
+
+  # Run the command in the background
+  exec "${@:-}" &
```

Comment:
This will do something very unexpected if `$@` is empty (`exec` without an argument just redirects a pipeline), so don't use the default parameter substitution; you'll want the script to fail in that case.

Reply:
Also, `exec` replaces the current shell. Nothing after this line will even run AFAIK!? I am confused.

Reply:
Oh wait, you `exec … &`, well, you can just drop `exec` altogether then and do `"$@" &`